File: weather.js
const express = require('express')
const axios = require('axios')

const app = express()

app.get('/weather', (req, res) => {
    const city = req.query.city
    if (!city) {
        return res.status(400).send('Please provide a city name in the "city" query parameter')
    }
    const apiKey = '081c7cbb1eb464e8c5bf244e8134ecfd' // replace with your API key
    const url = `http://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}`
    axios.get(url)
        .then(response => {
            res.send(response.data)
        })
        .catch(error => {
            res.status(error.response.status).send(error.response.data)
        })
})

app.listen(5000, () => {
    console.log('Server listening on port 5000')
})

File: weather.py
from flask import Flask, request
import requests

app = Flask(__name__)

@app.route('/weather', methods=['GET'])
def weather():
    city = request.args.get('city')
    if not city:
        return "Please provide a city name in the 'city' query parameter", 400
    api_key = "YOUR_API_KEY" # replace with your API key
    url = f"http://api.openweathermap.org/data/2.5/weather?q={city}&appid=081c7cbb1eb464e8c5bf244e8134ecfd"
    response = requests.get(url)
    if response.status_code != 200:
        return "Error: " + response.text, response.status_code
    return response.text

if __name__ == '__main__':
    app.run(debug=True)
